lib/view/templates.py

```diff
@@ -1855,38 +1814,135 @@ def turn_empty_to_none(ls):
+        )
+
+        if not weights:
+            return None
+
+        return weighted_avg(pcts, weights)
+
+    return usage_weighted_avg
+
+
+def create_quota_tps_subgroup(
+    type: Literal["Read"] | Literal["Write"],
+    key: Literal["read-info"] | Literal["write-info"],
+):
+    return Subgroup(
+        type,
+        (
+            Field(
+                "Quota",
+                Projectors.Func(
+                    FieldType.number,
+                    extract_value_from_dict("quota"),
+                    Projectors.Identity("data", key),
                 ),
-                # TODO: Support for Subgroups to have Subgroups to have Scan/Query be a
-                # subgroup of Write
-                Field(
-                    "Scan/Query Limited RPS",
-                    Projectors.Func(
-                        FieldType.undefined,
-                        extract_value_from_dict("scan-query-rps-limited"),
-                        Projectors.Identity("data", "write-info"),
+                converter=Converters.scientific_units,
+                aggregator=Aggregators.sum(converter=Converters.scientific_units),
+            ),
+            Field(
+                "Usage%",
+                Projectors.Div(
+                    Projectors.Sum(
+                        Projectors.Func(
+                            FieldType.number,
+                            extract_value_from_dict("single-record-tps"),
+                            Projectors.Identity("data", key),
+                        ),
+                        Projectors.Func(
+                            FieldType.number,
+                            extract_value_from_dict("scan-query-rps-limited"),
+                            Projectors.Identity("data", key),
+                        ),
                     ),
-                ),
-                Field(
-                    "Scan/Query Limitless",
                     Projectors.Func(
-                        FieldType.undefined,
-                        extract_value_from_dict("scan-query-limitless"),
-                        Projectors.Identity("data", "write-info"),
+                        FieldType.number,
+                        lambda x: _ignore_zero(extract_value_from_dict("quota")(x)),
+                        Projectors.Identity("data", key),
                     ),
                 ),
+                converter=Converters.ratio_to_pct,
+                aggregator=ComplexAggregator(
+                    create_quota_weighted_avg(type), converter=Converters.ratio_to_pct
+                ),
+                formatters=(
+                    Formatters.red_alert(lambda edata: edata.value * 100 >= 90.0),
+                    Formatters.yellow_alert(lambda edata: edata.value * 100 >= 75.0),
+                ),
+            ),
+            Field(
+                "Single Record TPS",
+                Projectors.Func(
+                    FieldType.undefined,
```

Comment:
Why are these undefined field types? I'd expect them to be numbers.

Reply:
good catch